src/loop.jl

```diff
@@ -128,35 +136,55 @@ function interval_loop(factory_like, model_kwargs::Dict,
                 if !isnothing(demand_flexibility.cost_up)
                     bus_demand_flex_cost_up = permutedims(
                         Matrix(
-                            demand_flexibility.cost_up[interval_start:interval_end, 2:end]
+                            demand_flexibility.cost_up[
+                                interval_start:interval_end, 2:end
+                            ]
                         )
                     )
                 end
                 if !isnothing(demand_flexibility.cost_dn)
                     bus_demand_flex_cost_dn = permutedims(
                         Matrix(
-                            demand_flexibility.cost_dn[interval_start:interval_end, 2:end]
+                            demand_flexibility.cost_dn[
+                                interval_start:interval_end, 2:end
+                            ]
                         )
                     )
                 end
-                for t in 1:interval, i in 1:length(sets.load_bus_idx)
```

Comment:
Aha, previously we used same variable `i` as the outer loop variable here, which will cause dead loop, right?

Reply:
Not in Julia. Just doing some basic testing, it appears that it handles this problem pretty well. If both an inner loop and outer loop use `i`, then the outer loop will pick back up after the inner loop completes. There would be an issue if you wanted to use both the outer loop and inner loop variables. There wasn't a problem here, I just changed it for readability (and correctness) purposes.

Reply:
Thanks for the explanation. That's good to know.